coprocess_id_extractor.go

```diff
@@ -164,62 +166,59 @@ type RegexExtractorConfig struct {
 }
 
 func (e *RegexExtractor) ExtractAndCheck(r *http.Request) (SessionID string, returnOverrides ReturnOverrides) {
-	var extractorOutput string
-	var config RegexExtractorConfig
-
-	err := mapstructure.Decode(e.Config.ExtractorConfig, &config)
-	if err != nil {
-		returnOverrides = e.Error(r, err, "Can't decode RegexExtractor configuration")
-		return SessionID, returnOverrides
+	// Parse specific configuration settings:
+	if e.cfg == nil {
+		if err := mapstructure.Decode(e.Config.ExtractorConfig, &e.cfg); err != nil {
```

Comment:
seems like taking address of nil pointer here, we don't need to take address anyways as `e.cfg` is already pointer

Reply:
https://github.com/mitchellh/mapstructure/blob/master/mapstructure.go#L194

Reply:
I mean you probably  wanted to do
```
if e.cfg == nil {
  config := &RegexExtractorConfig{}
  if err := mapstructure.Decode(e.Config.ExtractorConfig, config) {
  ...
  }
  e.cfg = config
...
```

Reply:
👍